### DISCUSSION TITLE
Bungeecord Cross-Server Economy

### DISCUSSION THREAD

**Discussion by LordGono** [Zrips/CMI#3858] [Feature Requests]
**Description of feature/improvement request:** I've been looking for days now for a method to use the CMI economy in a cross-server situation. As of right now, using the built-in MySQL feature I have managed to get players' economy data to travel with them when they switch from one server to another. The main issue is "offline" transactions. Whether it is the use of /pay or another shop plugin, if the player receiving the money is offline or in a different linked server, their balance will not be changed. It is my assumption that this is because economy based commands such as /pay or /money give do not write directly to the database, and instead stores the data locally until saved to the database during the save interval or force save on exit.

---
### Detailed explanation of what it should do, when and in what situation: ###
I'm looking for all player economy data to sync directly through the MySQL database. At the current moment, it appears that some economy data is handled locally. For example, a player on my network currently is on another server with a balance of $100. In the database it also shows him with $100, however, in my server which is also linked through the database, it shows him with $0, and if I pay him $5, it increases his balance on my server from $0 to $5, but does nothing to his balance on his server. I was wondering if there would be any way to make it so that it would be possible to fully sync these numbers cross-server and make it so I can also transfer money cross-server. If this is at all possible I would be insanely grateful.

### Basic explanation as to why this feature is needed and/or useful to have: ###
In a server network with multiple types of servers that all are meant to have a shared economy, this would help us greatly with actually syncing those balances instead of having to have 7+ separate balances with no easy way to transfer funds cross-server.

Finally, I wanted to thank you for your time for reading and any help you may be able to provide.

**Comment by ghost** [comment]
Yes we need this urgently. CMI can do so much, but with such an important feature it fails. I don't like installing Essentials and another bridge plugin. Please make this possible.
